equipment_handler/source/rtcmk.c

```diff
@@ -303,3 +280,572 @@ int RTCMK_ReadHours(/*I2C_TypeDef *i2c,*/
 
   return(ret);
 }
+
+/***************************************************************************//**
+ * @brief
+ *   Returns current content of hours register in decimal.
+ *
+ * @param[in] addr
+ *   I2C address, in 8 bit format, where LSB is reserved for R/W bit.
+ *
+ * @param[out] val
+ *   Reference to place result.
+ *
+ * @return
+ *   Returns 0 if registers written, <0 if unable to write to registers.
+ ******************************************************************************/
+int RTCMK_ReadWeek(uint8_t addr,
+                   uint8_t *val)
+{
+  int ret = -1;
+
+  uint8_t tmp = 0;
+
+  ret = RTCMK_RegisterGet(addr,RTCMK_RegWeek,&tmp);
+  if (ret < 0)
+  {
+    return(ret);
+  }
+
+  tmp &= _RTCMK_WEEK_WEEK_MASK;
+
+  *val = ((tmp & 0xF0) >> 4) * 10 + (tmp & 0x0F);
+
+  return(ret);
+}
+
+/***************************************************************************//**
+ * @brief
+ *   Returns current content of hours register in decimal.
+ *
+ * @param[in] addr
+ *   I2C address, in 8 bit format, where LSB is reserved for R/W bit.
+ *
+ * @param[out] val
+ *   Reference to place result.
+ *
+ * @return
+ *   Returns 0 if registers written, <0 if unable to write to registers.
+ ******************************************************************************/
+int RTCMK_ReadMonth(uint8_t addr,
+                    uint8_t *val)
+{
+  int ret = -1;
+
+  uint8_t tmp = 0;
+
+  ret = RTCMK_RegisterGet(addr,RTCMK_RegMonth,&tmp);
+  if (ret < 0)
+  {
+    return(ret);
+  }
+
+  tmp &= _RTCMK_MONTH_MONTH_MASK;
+
+  *val = ((tmp & 0xF0) >> 4) * 10 + (tmp & 0x0F);
```

Comment:
What is this doing and could we make it a macro

Reply:
I don't know what it does so I didn't want to touch it

Reply:
Testing can be accomplished with the ground station code. which I'm working on now

Reply:
The alarm functionality may or may not be used. but it could be tested with the scheduler if we decide to use it

Reply:
Where is it from then?